pkg/clusterresource/builder.go

```diff
@@ -444,17 +445,17 @@ func (o *Builder) generateInstallConfigSecret() (*corev1.Secret, error) {
 
 	// Remove osImage field from machinepool platform within installconfig.
 	if installConfig.Platform.Azure != nil {
-		ops := yamlpatch.Patch{
-			yamlpatch.Operation{
+		ops := []hivev1.PatchEntity{
+			{
 				Op:   "remove",
-				Path: yamlpatch.OpPath("/compute/0/platform/azure/osImage"),
+				Path: "/compute/0/platform/azure/osImage",
 			},
-			yamlpatch.Operation{
+			{
 				Op:   "remove",
-				Path: yamlpatch.OpPath("/controlPlane/platform/azure/osImage"),
+				Path: "/controlPlane/platform/azure/osImage",
 			},
 		}
-		modifiedBytes, err := ops.Apply(d)
+		modifiedBytes, err := yamlpatch.ApplyPatches(d, ops)
```

Comment:
Note that we're passing in JSON here (which is technically valid YAML) and getting back YAML.

Reply:
seems like ApplyPatches immediately turns it into JSON anyways

Reply:
Just ironic, no change required